app/controllers/summary_controller.rb

```diff
@@ -2,27 +2,38 @@ class SummaryController < ApplicationController
   before_action :authorize
 
   def index
-    client_namespace = current_user.client_slug.titleize
-    @client_summarizer = ClientSummarizer.new(
-      client_namespace: client_namespace,
-      fiscal_year: params[:fiscal_year]
-    )
-    @summary = @client_summarizer.run
+    client_namespaces = if current_user.gateway_admin?
+                          Proposal.client_slugs
+                        else
+                          [current_user.client_slug]
+                        end
+                        .map(&:titleize)
+
+    @summaries = client_namespaces.map do |cn|
+      get_client_summary(cn, params[:fiscal_year])
+    end
   end
 
   private
 
   def authorize
-    if !adminish_role? || !client_slug?
+    if !adminish_role? || (!client_slug? && !current_user.gateway_admin?)
```

Comment:
Not quite - the previous logic says that admins need to have an adminish role _and_ a client slug, whereas in the new logic, gateway admins don't need a client slug.

However, there's probably a clearer way to write this. I'll give it a go.